api/src/integrations/core-documents/base-record-utils.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const mongoose = require('mongoose');
+
+const defaultLog = require('../../utils/logger')('core-base-record-utils');
+const RecordController = require('../../controllers/record-controller');
+
+/**
+ * Core base record type handler that can be used directly, or extended if customizations are needed.
+ *
+ * Must contain the following functions:
+ * - transformRecord: (object) => object
+ * - saveRecord: (object) => any
+ *
+ * @class BaseRecordUtils
+ */
+class BaseRecordUtils {
+  /**
+   * Creates an instance of BaseRecordUtils.
+   *
+   * @param {*} auth_payload user information for auditing
+   * @param {*} recordType an item from record-type-enum.js -> RECORD_TYPE
+   * @memberof BaseRecordUtils
+   */
+  constructor(auth_payload, recordType) {
+    if (!recordType) {
+      throw Error('BaseRecordUtils - required recordType must be non-null.');
+    }
+
+    this.auth_payload = auth_payload;
+    this.recordType = recordType;
+  }
+
+  /**
+   * Transform an Core record into a NRPTI record.
+   *
+   * Note: Only transforms common fields found in ALL supported core import types.
+   *       To include other values, extend this class and adjust the object returned by this function as needed.
+   *
+   * @param {object} coreRecord Core record (required)
+   * @returns {object} NRPTI record.
+   * @throws {Error} if record is not provided.
+   * @memberof BaseRecordUtils
+   */
+  transformRecord(coreRecord) {
+    if (!coreRecord) {
+      throw Error('transformRecord - required coreRecord must be non-null.');
+    }
+
+    return {
+      _schemaName: this.recordType._schemaName,
+
+      addedBy: this.auth_payload.displayName,
+      updatedBy: this.auth_payload.displayName,
+
+      dateAdded: new Date(),
+      dateUpdated: new Date(),
```

Comment:
Similar to the above comment: something that was missed before, and is fixed in a few newer spots (the csv imports I tihnk?).

The `dateAdded/addedBy` should really just get set in the createRecord() function, and then similarly for the `dateUpdated/updatedBy` fields in the updateRecord() function.

Reply:
Updated